### ISSUE TITLE
`benchmark()` does not save models with `store_models = TRUE`?

### ISSUE THREAD

**Issue by pat-s** [mlr-org/mlr3#242]
``` r
library(mlr3)

tasks = mlr_tasks$mget(c("iris", "sonar"))
learners = mlr_learners$mget(c("classif.featureless", "classif.rpart"))
resamplings = mlr_resamplings$mget("holdout")

design = expand_grid(tasks, learners, resamplings)
print(design)
#>             task                     learner          resampling
#> 1: <TaskClassif> <LearnerClassifFeatureless> <ResamplingHoldout>
#> 2: <TaskClassif>       <LearnerClassifRpart> <ResamplingHoldout>
#> 3: <TaskClassif> <LearnerClassifFeatureless> <ResamplingHoldout>
#> 4: <TaskClassif>       <LearnerClassifRpart> <ResamplingHoldout>

set.seed(123)
bmr = benchmark(design, ctrl = mlr_control(store_model = TRUE))
#> INFO  [22:18:50.363] Benchmarking 4 experiments 
#> INFO  [22:18:50.385] Running learner 'classif.featureless' on task 'iris' (iteration 1/1)' 
#> INFO  [22:18:50.409] Running learner 'classif.rpart' on task 'iris' (iteration 1/1)' 
#> INFO  [22:18:50.434] Running learner 'classif.featureless' on task 'sonar' (iteration 1/1)' 
#> INFO  [22:18:50.442] Running learner 'classif.rpart' on task 'sonar' (iteration 1/1)' 
#> INFO  [22:18:50.473] Finished benchmark

# should store the model?
bmr$aggregated()$resample_result$experiment$model
#> NULL
```

<sup>Created on 2019-06-11 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by mllg** [comment]
The last line should be

```r
bmr$aggregated()$resample_result[[1]]$experiment(1)$model
```
And unfortunatly there is not much we can do about the missing error message: `bmr$aggregated()$resample_result` is a list, and accessing non-existing elements by name yields `NULL` in R.

**Comment by susanne-207** [comment]
If I train multiple learners on multiple tasks using `benchmark()` and setting `store_models = TRUE`,  not all fitted models are stored. 
Here is an example: 
```
library(mlr3)
library(mlr3learners)

## Benchmark: 
task_list = list(mlr_tasks$get("iris"), mlr_tasks$get("wine"))
learners = list(
    lrn("classif.ranger"), 
    lrn("classif.svm"), 
    lrn("classif.log_reg")
) 
design = benchmark_grid(
    tasks = task_list,
    learners = learners,
    resamplings = rsmp("holdout", ratio = 2/3)
) 
bmr = mlr3::benchmark(design, store_model = TRUE)

# only the learners for one task are stored
length(bmr$learners$learner)

# predict only works for the first task
bmr$learners$learner[[1]]$predict(task_list[[1]])
bmr$learners$learner[[2]]$predict(task_list[[1]])
bmr$learners$learner[[3]]$predict(task_list[[1]])

# this gives an error
bmr$learners$learner[[1]]$predict(task_list[[2]]) 
```

**Comment by mllg** [comment]
The learners returned by `learners` are selected as trained "representatives" of the base learners. You get the error because you have retrieved a learner trained on task 1, and tried to apply it on task 2.


This is confusing and fixed in 446353d95099250c4de3934d3ea37538fc9d2735. If you want to access a model from a trained learner, you need to select the learner from the table returned by `$score()` (select via task id, learner id and resampling iteration).
